frontend/src/forms/MonsterGallery.js

```diff
@@ -1,55 +1,48 @@
-import React from 'react';
-
-export default function MonsterGallery({ monsterImage }) {
-  const imageToDisplay = !monsterImage
-    ? '/monsterImages/monster0.jpg'
-    : monsterImage;
-
-  const numberOfDisplayedImage = parseInt(imageToDisplay.slice(-5, -4));
-
-  const amount = 5;
-  let images = [];
-  let currentImage = numberOfDisplayedImage;
-
-  function initImages() {
-    for (let i = 0; i < amount; i++) {
-      images[i] = '/monsterImages/monster' + i + '.jpg';
-    }
-  }
-  initImages();
-
-  function setImage() {
-    const currentlyDisplayedImage = window.document.getElementById(
-      'currentImage'
-    );
-    currentlyDisplayedImage.src = images[currentImage];
-  }
+import React, { useEffect, useState } from 'react';
+
+export default function MonsterGallery({
+  savedMonsterImage,
+  imageChangedHandler,
+}) {
+  const images = [
+    '/monsterImages/monster0.jpg',
+    '/monsterImages/monster1.jpg',
+    '/monsterImages/monster2.jpg',
+    '/monsterImages/monster3.jpg',
+    '/monsterImages/monster4.jpg',
+  ];
```

Comment:
Can be moved outside of `MonsterGallery` so it is not redefined on every update cycle, just a small performance improvement..